AK/ByteBuffer.h

```diff
@@ -117,10 +117,24 @@ class ByteBuffer {
     [[nodiscard]] bool is_empty() const { return m_size == 0; }
     [[nodiscard]] size_t size() const { return m_size; }
 
-    [[nodiscard]] u8* data() { return m_inline ? m_inline_buffer : m_outline_buffer; }
+#ifdef AK_COMPILER_GCC
+#    pragma GCC diagnostic push
+//   Workaround for https://gcc.gnu.org/bugzilla/show_bug.cgi?id=109727
+#    pragma GCC diagnostic ignored "-Warray-bounds"
+#endif
+    [[nodiscard]] u8* data()
+    {
+        return m_inline ? m_inline_buffer : m_outline_buffer;
+    }
     [[nodiscard]] u8 const* data() const { return m_inline ? m_inline_buffer : m_outline_buffer; }
+#ifdef AK_COMPILER_GCC
+#    pragma GCC diagnostic pop
+#endif
 
-    [[nodiscard]] Bytes bytes() { return { data(), size() }; }
+    [[nodiscard]] Bytes bytes()
+    {
+        return { data(), size() };
+    }
```

Comment:
cc @rymiel on the odd formatting change here, but I suspect it's related to `ColumnLimit: 0` 🤔. Or maybe it just makes different decisions after a pragma.

Reply:
Yes, it appears like since the pragma forces a newline after it, it makes it think the definition isn't on one line (is my guess).
I'm not sure if it's been reported before, it feels like it should be, but I don't remember seeing it before. Definitely a bug though.

Reply:
(just as closure, it's now fixed on trunk: https://github.com/llvm/llvm-project/commit/06763ea5d8f96625545bb2c2445363aca9922bf1, but trunk is a long time away from being widely usable)